inference-engine/tests/functional/inference_engine/serialization/single_layer/pooling.cpp

```diff
@@ -0,0 +1,89 @@
+
+using namespace LayerTestsDefinitions;
+
+namespace {
+    TEST_P(PoolingLayerTest, Serialize) {
+        Serialize();
+    }
+
+    const std::vector<InferenceEngine::Precision> netPrecisions = {
+            InferenceEngine::Precision::FP32,
+            InferenceEngine::Precision::FP16
+    };
+
+    /* ============= POOLING ============= */
+    const std::vector<std::vector<size_t >> kernels = {{3, 3},
+                                                   {3, 5}};
+    const std::vector<std::vector<size_t >> strides = {{1, 1},
+                                                    {1, 2}};
+    const std::vector<std::vector<size_t >> padBegins = {{0, 0},
+                                                        {0, 2}};
+    const std::vector<std::vector<size_t >> padEnds = {{0, 0},
+                                                    {0, 2}};
+
+    const std::vector<ngraph::op::RoundingType> roundingTypes = {
+       ngraph::op::RoundingType::FLOOR,
+       ngraph::op::RoundingType::CEIL
+    };
+
+    const std::vector<ngraph::op::PadType> padTypes = {
+        ngraph::op::PadType::EXPLICIT,
+        ngraph::op::PadType::SAME_UPPER,
+        ngraph::op::PadType::VALID
+    };
+
+    /* ============= AVERAGE POOLING ============= */
+
+    const auto avgExcludePadParams = ::testing::Combine(
+            ::testing::Values(ngraph::helpers::PoolingTypes::AVG),
+            ::testing::ValuesIn(kernels),
+            ::testing::ValuesIn(strides),
+            ::testing::ValuesIn(padBegins),
+            ::testing::ValuesIn(padEnds),
+            ::testing::ValuesIn(roundingTypes),
+            ::testing::ValuesIn(padTypes),
+            ::testing::Values(true));
+
+    INSTANTIATE_TEST_CASE_P(smoke_AvgPoolExcluePad, PoolingLayerTest,
+            ::testing::Combine(
+                avgExcludePadParams,
+                ::testing::ValuesIn(netPrecisions),
+                ::testing::Values(InferenceEngine::Precision::UNSPECIFIED),
+                ::testing::Values(InferenceEngine::Precision::UNSPECIFIED),
+                ::testing::Values(InferenceEngine::Layout::ANY),
+                ::testing::Values(InferenceEngine::Layout::ANY),
+                ::testing::Values(std::vector<size_t>{511, 11, 13, 15}),
+                ::testing::Values(CommonTestUtils::DEVICE_CPU)),
+            PoolingLayerTest::getTestCaseName);
+
+    const auto avgPadParams = ::testing::Combine(
+            ::testing::Values(ngraph::helpers::PoolingTypes::AVG),
+            ::testing::ValuesIn(kernels),
+            ::testing::ValuesIn(strides),
+            ::testing::ValuesIn(padBegins),
+            ::testing::ValuesIn(padEnds),
+            ::testing::ValuesIn(roundingTypes),
+            ::testing::ValuesIn(padTypes),
+            ::testing::Values(false));
+
+    INSTANTIATE_TEST_CASE_P(smoke_AvgPool, PoolingLayerTest,
+            ::testing::Combine(
+                avgPadParams,
+                ::testing::ValuesIn(netPrecisions),
```

Comment:
Why do we need all this IE things in test where we work only with nGraph?

Reply:
We use test classes for single layer tests which requires this information.

Reply:
We use test classes for single layer tests which required this information

Reply:
@GlebKazantaev the idea behind Serialization SLT is to reuse models from regular 'CompareRef' SLT. [Here](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/shared_test_classes/src/single_layer/pooling.cpp#L12) is the test class for Pooling and it already contains dependencies on IE types (e.g. IE::Precision).